pyEpiabm/pyEpiabm/tests/test_unit/test_sweep/test_spatial_sweep.py

```diff
@@ -14,69 +15,134 @@ class TestSpatialSweep(TestMockedLogs):
+        self.microcell_susc = self.cell_susc.microcells[0]
+
+        # self.cell_non_susc.add_microcells(1)
+        # self.microcell_non_susc = self.cell_non_susc.microcells[0]
+
         self.microcell_inf.add_people(100)
         self.infector = self.microcell_inf.persons[0]
+        self.infector.update_status(InfectionStatus.InfectMild)
+
+        self.microcell_susc.add_people(1)
+        self.infectee = self.microcell_susc.persons[0]
+
+        # self.microcell_non_susc.add_people(1)
+        # self.non_infectee = self.microcell_non_susc.persons[0]
+        # self.non_infectee.update_status(InfectionStatus.Recovered)
+
         self.infector.infectiousness = 1.0
         Parameters.instance().time_steps_per_day = 1
         Parameters.instance().do_CovidSim = False
 
+        # 2nd population
+        # Only has a single cell with a single person in
+        # Used for checking exceptions in the code
+        self.pop_only1 = Population()
+        self.pop_only1.add_cells(1)
+        self.cell_inf_only1 = self.pop_only1.cells[0]
+
+        self.cell_inf_only1.add_microcells(1)
+        self.microcell_inf_only1 = self.cell_inf_only1.microcells[0]
+
+        self.microcell_inf_only1.add_people(100)
+        self.infector_only1 = self.microcell_inf_only1.persons[0]
+
+        self.infector_only1.infectiousness = 1.0
+        Parameters.instance().time_steps_per_day = 1
+        Parameters.instance().do_CovidSim = False
+
+        # 3rd Population
+        # 2 cells each with a single microcell with a single person
+        # 1 cell has a mildly infected individual
+        # the other has a recovered individual
+        self.pop_no_infectees = Population()
+        self.pop_no_infectees.add_cells(2)
+        self.cell_no_infectees_inf = self.pop_no_infectees.cells[0]
+        self.cell_no_infectees_rec = self.pop_no_infectees.cells[1]
+
+        self.cell_no_infectees_inf.add_microcells(1)
+        self.microcell_no_infectees_inf = self.\
+            cell_no_infectees_inf.microcells[0]
+
+        self.cell_no_infectees_rec.add_microcells(1)
+        self.microcell_no_infectees_rec = self.\
+            cell_no_infectees_rec.microcells[0]
+
+        self.microcell_no_infectees_inf.add_people(1)
+        self.no_infectees_infector = self.microcell_no_infectees_inf.persons[0]
+        self.no_infectees_infector.update_status(InfectionStatus.InfectMild)
+
+        self.microcell_no_infectees_rec.add_people(1)
+        self.no_infectees_rec = self.microcell_no_infectees_rec.persons[0]
+        self.no_infectees_rec.update_status(InfectionStatus.Recovered)
+
+    @mock.patch("pyEpiabm.utility.DistanceFunctions.dist_euclid")
+    def test_near_neighbour(self, mock_dist):
+        Parameters.instance().infection_radius = 1000
+        test_pop = self.pop
+        test_sweep = SpatialSweep()
+        mock_dist.return_value = 2
+        test_sweep.bind_population(test_pop)
+        self.assertEqual(self.cell_inf.nearest_neighbours, {1: 2})
```

Comment:
Great testing! But as there are two functions (near_neighbour and bind_population), I think it's a good idea to create one for test_near_neighbour with given cell and other_cell, which only tests if the dictionaries are equal. Then create a test_bind_population function with given population, which calls the bind_population function for testing. It will be more clear!